tests/unit-tests/dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -1414,6 +1454,49 @@ TEST(DDS_DCPS_XTypes_DynamicData, Appendable_StructWithOptionalMembers)
   verify_index_mapping(data);
 }
 
+TEST(DDS_DCPS_XTypes_DynamicData, Appendable_StructWithOptionalMembersXCDR1)
+{
+  const XTypes::TypeIdentifier& ti = DCPS::getCompleteTypeIdentifier<DCPS::AppendableSingleValueStruct_xtag>();
+  const XTypes::TypeMap& type_map = DCPS::getCompleteTypeMap<DCPS::AppendableSingleValueStruct_xtag>();
+  const XTypes::TypeMap::const_iterator it = type_map.find(ti);
+  EXPECT_TRUE(it != type_map.end());
+
+  XTypes::TypeLookupService tls;
+  tls.add(type_map.begin(), type_map.end());
+  XTypes::CompleteTypeObject cto = it->second.complete;
+  cto.struct_type.member_seq[2].common.member_flags |= XTypes::IS_OPTIONAL;
+  cto.struct_type.member_seq[5].common.member_flags |= XTypes::IS_OPTIONAL;
+  XTypes::DynamicType_rch dt = tls.complete_to_dynamic(cto, DCPS::GUID_t());
+
+  unsigned char single_value_struct[] = {
+    0x00,0x00,0x00,0x03, // +4=4 my_enum
+    0x00,0x00,0x00,0x0a, // +4=8 int_32
+    0x00, // +1=9 Omitting uint_32
```

Comment:
Yes, I forgot about this. You can also see 7.4.1.2 for a text description of this. XCDR1 optional in final and appendable requires an isolated XCDR1 parameter id instead of a boolean. `read_parameter_id` would be able to do it as it's a simple case. However I agree that I don't think this PR needs to support it.